processing/src/main/java/io/druid/segment/IndexMergerV9.java

```diff
@@ -644,7 +644,7 @@ private void mergeIndexesAndWriteColumns(
         if (dimensionSkipFlag.get(i)) {
           continue;
         }
-        if (dims[i] == null || dims[i].length == 0) {
+        if (dims[i] == null || dims[i].length == 0 || dims[i][0] == 0) {
```

Comment:
Looks like it is guaranteed due to the block below. but, the other check should still be modified, I think.